xray/plotting/plotting.py

```diff
@@ -441,3 +441,69 @@ def plot_pcolormesh(x, y, z, ax, **kwargs):
     ax.set_ylim(y[0], y[-1])
 
     return ax, primitive
+
+
+class _PlotMethods(object):
+    '''
+    Plotmethods class
+    '''
+
+    # Basically this is a bunch of hacking to put in the dataarray as the
```

Comment:
The current implementation works, but I'm not happy with it. The biggest problem is the way all the 2d plots are added- they all have the same form, so should be added in a more DRY way.